native/cocos/editor-support/spine-wasm/wasmSpineExtension.cpp

```diff
@@ -29,44 +30,23 @@ char *WasmSpineExtension::_readFile(const String &path, int *length) {
 void *WasmSpineExtension::_alloc(size_t size, const char *file, int line) {
     SP_UNUSED(file);
     SP_UNUSED(line);
-
-    if (size == 0)
-        return 0;
-    void *ptr = new uint8_t[size];
-    return (void *)ptr;
+    return ::malloc(sizeof(uint8_t) * size);
 }
 
 void *WasmSpineExtension::_calloc(size_t size, const char *file, int line) {
     SP_UNUSED(file);
     SP_UNUSED(line);
-
-    if (size == 0)
-        return 0;
-    uint8_t *ptr = new uint8_t[size];
-    if (ptr) memset(ptr, 0, size);
-    return (void *)ptr;
+    return ::calloc(1, size);
```

Comment:
Why not check size before calloc?

Reply:
In fact, malloc/calloc support to access `zero` value to size parameter. But it's an undefined behavior in c standard.
https://en.cppreference.com/w/c/memory/malloc
> If size is zero, the behavior of malloc is implementation-defined. For example, a null pointer may be returned. Alternatively, a non-null pointer may be returned; but such a pointer should not be [dereferenced](https://en.cppreference.com/w/c/language/operator_member_access), and should be passed to [free](https://en.cppreference.com/w/c/memory/free) to avoid memory leaks.

Emscripten uses `dlmalloc` to implement c standard malloc/calloc/free  by default. dlmalloc will return a small chunk of memory if passing 0 to it.

Yes, to avoid the undefined behaviior, I think adding back the `zero check` is ok.

Reply:
Reverted.